lib/rule_engine/utils.py

```diff
@@ -0,0 +1,38 @@
+import datetime
+import re
+
+
+_sub_regex = r'[0-9]+([,.][0-9]+)?'
+timedelta_regex = (
+    r'P(?!\b)'
+    f'(?P<weeks>{_sub_regex}W)?'
+    f'(?P<days>{_sub_regex}D)?'
+    r'(T'
+    f'(?P<hours>{_sub_regex}H)?'
+    f'(?P<minutes>{_sub_regex}M)?'
+    f'(?P<seconds>{_sub_regex}S)?'
+    r')?'
+)
+timedelta_re = re.compile(f"^{timedelta_regex}$")
```

Comment:
Any reason not to make this case insensitive?

```suggestion
timedelta_re = re.compile(f"^{timedelta_regex}$", flags=re.IGNORECASE)
```

Reply:
To confirm, this is case-sensitive because the ISO 8601 spec is case-sensitive.